src/Compilers/CSharp/Portable/Symbols/SubstitutedNamedTypeSymbol.cs

```diff
@@ -162,14 +162,6 @@ internal override ImmutableArray<NamedTypeSymbol> GetInterfacesToEmit()
             throw ExceptionUtilities.Unreachable;
         }
 
-        protected sealed override ImmutableArray<NamedTypeSymbol> MakeAllInterfaces()
-        {
-            // Because declared types will have been checked for "uniqueness of implemented interfaces" (C# 4 spec, 13.4.2),
-            // we are guaranteed that none of these substitutions collide in a correct program.  Consequently, we can simply
-            // substitute the original interfaces.
-            return _unbound ? ImmutableArray<NamedTypeSymbol>.Empty : Map.SubstituteNamedTypes(OriginalDefinition.AllInterfacesNoUseSiteDiagnostics);
-        }
```

Comment:
The comment was incorrect. Section 13.4.2 in the C# 6 spec actually gives an example where a generic Derived type re-implements an interface. 
The simple logic of taking all interfaces from the original definition and applying type substitution to those yields duplicate interfaces in this case. So we need to rely on the brute force approach inherited from TypeSymbol.MakeAllInterfaces().